### PR TITLE
Feature/57347 add validation connection check unexpected content for nextcloud storages

### PR DESCRIPTION
**Pull request by apfohl**
# Ticket
https://community.openproject.org/projects/openproject/work_packages/57347/github

# What are you trying to accomplish?
Adding a validator for the `NextcloudConnectionValidation` that checks for unforseen folder contents inside the group folder on Nextcloud side.

# What approach did you choose and why?
I just programmed it!

# Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation in Lookbook (patterns, previews, etc)
- [ ] Tested major browsers (Chrome, Firefox, Edge, ...)

### CODE REVIEW

File: modules/storages/app/common/storages/peripherals/nextcloud_connection_validator.rb [hunk 1]
```diff
@@ -59,6 +60,17 @@ def query
                      .call(storage: @storage, auth_strategy:)
       end
 
+      def files
```
Comment (Kharonus) [inline modules/storages/app/common/storages/peripherals/nextcloud_connection_validator.rb:68]: 🟡 minor, but if we have now a meaningful name to distinguish outbound requests, can we rename the other query to `capabilities`?

File: modules/storages/app/common/storages/peripherals/nextcloud_connection_validator.rb [hunk 2]
```diff
@@ -59,6 +60,17 @@ def query
...
+        @files ||= Peripherals::Registry
+                     .resolve("#{@storage.short_provider_type}.queries.files")
+                     .call(
+                       storage: @storage,
+                       auth_strategy: Peripherals::Registry
```
Comment (Kharonus) [inline modules/storages/app/common/storages/peripherals/nextcloud_connection_validator.rb]: 🟡 also minor, can we extract the userless auth strategy to a private method, too.

naming the current `noop`, and this one `userless`. IMHO a small improvement of readability

File: modules/storages/config/locales/en.yml [hunk 3]
```diff
@@ -217,7 +217,9 @@ en:
         placeholder: Check your connection against the server.
         subtitle: Connection validation
         tenant_id_wrong: The configured directory (tenant) id is invalid. Please check the configuration.
-        unexpected_content: Unexpected content found in the drive.
+        unexpected_content:
+          nextcloud: Unexpected content found in the managed folder.
```
Comment (Kharonus) [inline modules/storages/config/locales/en.yml]: 🟡 not sure about the AC of this feature, but should we use the wording `in the managed group folder`? we do use it elsewhere, too.

File: modules/storages/spec/common/storages/peripherals/nextcloud_connection_validator_spec.rb [hunk 4]
```diff
@@ -141,11 +150,42 @@
         end
       end
     end
+
+    context "if the request returns unexpected files" do
```
Comment (Kharonus) [inline modules/storages/spec/common/storages/peripherals/nextcloud_connection_validator_spec.rb]: 🔴 missing testcase, what happens if the `files` query returns a failure (not_found can be a good example). This opens then the question, is this maybe even another error? a la: "group folder not set up". Maybe we need an opinion from dominic here.

**Review by Kharonus — CHANGES_REQUESTED** [review]
I do have one bigger and a couple of very small requests.

File: modules/storages/spec/common/storages/peripherals/nextcloud_connection_validator_spec.rb (hunk 4 above)
Comment (apfohl) [reply modules/storages/spec/common/storages/peripherals/nextcloud_connection_validator_spec.rb]: I will integrate a test for the failure case for now, and we will discuss this case later.

File: modules/storages/app/common/storages/peripherals/nextcloud_connection_validator.rb [hunk 5]
```diff
@@ -45,6 +45,7 @@ def validate
           .or { missing_dependencies }
           .or { version_mismatch }
           .or { request_failed_with_unknown_error }
+          .or { with_unexpected_content }
```
Comment (Kharonus) [inline modules/storages/app/common/storages/peripherals/nextcloud_connection_validator.rb:48]: 🟡 I think, we need to change the order here. first check for unexpected content, then unknown errors. As the unknown errors should catch everything, that is not handled by the previous rules.

**Pull request merged**